Dockerfile

```diff
@@ -90,20 +93,42 @@ RUN pip3 install -r requirements.txt -c constraints.txt
 RUN gem install bundler && bundle install
 RUN apt-get remove -y build-essential ruby-dev
 
+# nginx unprivileged setup
+RUN ln -sf /dev/stdout /var/log/nginx/access.log && \
+    ln -sf /dev/stderr /var/log/nginx/error.log && \
+    sed -i '/user www-data;/d' /etc/nginx/nginx.conf && \
+    sed -i 's,/run/nginx.pid,/tmp/nginx.pid,' /etc/nginx/nginx.conf && \
+    sed -i "/^http {/a \    proxy_temp_path /tmp/proxy_temp;\n    client_body_temp_path /tmp/client_temp;\n    fastcgi_temp_path /tmp/fastcgi_temp;\n    uwsgi_temp_path /tmp/uwsgi_temp;\n    scgi_temp_path /tmp/scgi_temp;\n" /etc/nginx/nginx.conf && \
+    mkdir -p /var/cache/nginx && \
+    chown -R www-data:0 /var/cache/nginx && \
+    chmod -R g+w /var/cache/nginx
+
 RUN mkdir -p tmp && \
     echo "UPLOAD_DIR = '$PWD/tmp'" > at/config.py && \
     echo "VERSION = '${VERSION}'" >> at/config.py && \
     echo "REQUIRE_AUTH = False" >> at/config.py && \
     echo "DT_LATEST_DRAFT_URL = 'https://datatracker.ietf.org/api/rfcdiff-latest-json'" >> at/config.py && \
     echo "ALLOWED_DOMAINS = ['ietf.org', 'rfc-editor.org', 'github.com', 'githubusercontent.com', 'github.io', 'gitlab.com', 'gitlab.io', 'codeberg.page']" >> at/config.py && \
-    python3 version.py >> at/config.py
+    python3 version.py >> at/config.py && \
+    chown -R www-data:0 /usr/src/app/tmp
+
+# cache configuration
+RUN mkdir -p /tmp/cache/xml2rfc && \
+    mkdir -p /tmp/cache/refcache && \
+    ln -sf /tmp/cache/xml2rfc /var/cache/xml2rfc && \
+    chown -R www-data:0 /tmp/cache
```

Comment:
Would you explain what the symlink is for?

Reply:
symlink is for the `xml2rfc` cache. By default `xml2rfc` looks at `/var/cache/xml2rfc` first.